File: vue/src/api/usermanager/usergroup.js
import request from "@/utils/request";

// 获取所有用户keyname
export function getAllUserKeyName() {
  return request({
    url: "/alluser/keyname",
    method: "post"
  });
}

// 获取所有用户组
export function getUserGroups() {
  return request({
    url: "/usergroup/list",
    method: "post"
  });
}

// 修改用户组
export function updateUserGroup(data) {
  return request({
    url: "/usergroup/update",
    method: "post",
    data
  });
}

// 创建用户组
export function createUserGroups(data) {
  return request({
    url: "/usergroup/create",
    method: "post",
    data
  });
}

// 删除用户组
export function deleteUserGroups(id) {
  return request({
    url: `/usergroup/delete?id=${id}`,
    method: "get"
  });
}

File: go/routegroup/usermanager/usergroup.go
package usermanager

import (
	"itflow/handle/usercontroller/usergroup"
	"itflow/model"

	"github.com/hyahm/xmux"
)

var UserGroupPage *xmux.RouteGroup

func init() {
	UserGroupPage = xmux.NewRouteGroup()
	// 获取所有用户的信息关系
	UserGroupPage.Post("/alluser/keyname", usergroup.GetAllUserKeyName)
	// 获取所有用户组
	UserGroupPage.Post("/usergroup/list", usergroup.List)
	// 修改所有用户组
	UserGroupPage.Post("/usergroup/update", usergroup.Update).BindJson(&model.UserGroup{})
	// // 增加所有用户组
	UserGroupPage.Post("/usergroup/create", usergroup.Create).BindJson(&model.UserGroup{})
	// // 删除所有用户组
	UserGroupPage.Get("/usergroup/delete", usergroup.Delete)

}
